rest/diagnostic_doc_api.go

```diff
@@ -39,3 +54,53 @@ func (h *handler) handleGetDocChannels() error {
 	h.writeJSON(resp)
 	return nil
 }
+
+// HTTP handler for running a document through the sync function and returning the results
+func (h *handler) handleSyncFnDryRun() error {
+	docid := h.getQuery("docid")
```

Comment:
The spec has docid on the path, and not as a query string parameter.  I guess this is an intentional change to support running without specifying a docid?  Should the spec get updated?  
I'm not exactly clear on the intended behaviour in the following combinations of input - can we document those in the comment for this function:
- doc id only, no rev, no body
- body only 
- doc id and body
- doc id, rev, body
- rev only?